src/Tribe/Views/V2/Views/List_Behavior.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Provides methods for Views that behave like a List View.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views
+ */
+
+namespace Tribe\Events\Views\V2\Views;
+
+use Tribe__Context as Context;
+use Tribe__Date_Utils as Dates;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Trait List_Behavior
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Views
+ *
+ * @property Context $context The current View context.
+ */
+trait List_Behavior {
+
+	/**
+	 * Set up the datepicker-related template vars for the View.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $template_vars The current template vars.
+	 *
+	 * @return array The template vars, modified to add the required variables.
+	 */
+	protected function setup_datepicker_template_vars( array $template_vars ) {
+		if ( ! isset( $this->context ) ) {
+			return $template_vars;
+		}
+
+		$now                 = Dates::build_date_object( Arr::get( $template_vars, 'now', 'now' ) );
+		$today               = Dates::build_date_object( Arr::get( $template_vars, 'today', 'today' ) );
+		// This could yield an empty string, that we want to discard to keep "now" if that's the case.
+		$bar_date            = Arr::get( $template_vars, [ 'bar', 'date' ]) ?: $now;
```

Comment:
But also, related to `?:` usage here, it will just return `true` when the conditional is `true`. Is that what we want?
```suggestion
		$bar_date            = Arr::get( $template_vars, [ 'bar', 'date' ] ) ?: $now;
```

Reply:
The `?:` is a short-hand for the `!empty($v) ? $v : $k` construct.
If the return value of the `Arr::get()` call is empty or is not set, then it will return `$now`.
It's what we want.